com.unity.render-pipelines.universal/Editor/Converter/ReadonlyMaterialConverter.cs

```diff
@@ -0,0 +1,279 @@
+            {"Default-ParticleSystem", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/ParticlesUnlit.mat"},
+            {"Default-Particle", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/ParticlesUnlit.mat"},
+            {"Default-Terrain-Diffuse", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/TerrainLit.mat"},
+            {"Default-Terrain-Specular", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/TerrainLit.mat"},
+            {"Default-Terrain-Standard", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/TerrainLit.mat"},
+            {"Sprites-Default", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Sprite-Unlit-Default.mat"},
+            {"Sprites-Mask", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Sprite-Unlit-Default.mat"},
+
+            // These are custom URP shaders right now, but the old BiRP ones do render in Unity. Do we need to convert?
+            {"SpatialMappingOcclusion", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/SpatialMappingOcclusion.mat"},
+            {"SpatialMappingWireframe", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/SpatialMappingWireframe.mat"},
+
+            // TODO: Replace these with something more appropriate
+            {"Default UI Material", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Lit.mat"},
+            {"ETC1 Supported UI Material", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Lit.mat"},
+            {"Default-Line", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Lit.mat"},
+            {"Default-Skybox", "Packages/com.unity.render-pipelines.universal/Runtime/Materials/Lit.mat"},
+        };
+    }
+
+    internal class ReadonlyMaterialConverter : RenderPipelineConverter
+    {
+        public override string name => "Readonly Material Converter";
+        public override string info => "Converts references to Built-In readonly materials to URP readonly materials. This will create a .index file and that can take a long time.";
+        public override Type container => typeof(BuiltInToURPConverterContainer);
+        public override bool needsIndexing => true;
+
+        public override void OnInitialize(InitializeConverterContext ctx, Action callback)
+        {
+            using (var context = Search.SearchService.CreateContext("asset", "urp:convert-readonly"))
+            {
+                Search.SearchService.Request(context,  (c, items) =>
+                {
+                    // we're going to do this step twice in order to get them ordered, but it should be fast
+                    var orderedRequest = items.OrderBy(req =>
+                    {
+                        GlobalObjectId.TryParse(req.id, out var gid);
+                        return gid.assetGUID;
+                    })
+                        .ToList();
```

Comment:
You do not need to convert to a list since you are looping orderedRequest enumerator right away in the foreach below.

Reply:
done